### ISSUE TITLE
textinput: focus ring not visible when container has background

### ISSUE THREAD

**Issue by KaidenR** [pluralsight/design-system#283]
### Expected behavior

When focused, the user should be able to see a blue focus ring around the input

### Actual behavior

If an ancestor element of the input has a background, the focus ring is z-indexed behind the background and therefore doesn't show up.

I took the screenshot below of the textinput in storybook after adding a div around it with a green background. You can see the focus ring poking out behind the green background on the left side.

![image](https://user-images.githubusercontent.com/679346/46984227-6786ce00-d0a1-11e8-85e3-3740e21b75ac.png)

It looks like this can be fixed by just giving the component its own stacking context (give the `.fieldContainer` div a z-index of 0). This will cause the -1 and -2 z-indexes in the focus ring pseudo-elements to be relative to the component instead of relative to the rest of the app.

### Steps to repro

1. Add a textinput inside a container element with a background.
2. Focus the input - verify no focus ring shows up.

**Comment by jaketrent** [comment]
Hey Kaiden, yes, this is a stumbling block we've seen a couple times with the focus indicator.  For now, please create a new stacking context as you've discovered.

For more details, please see: https://github.com/pluralsight/design-system/issues/267#issuecomment-426027989
